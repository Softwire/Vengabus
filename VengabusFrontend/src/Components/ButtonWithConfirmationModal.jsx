import React from 'react';
import { Modal, Alert, Button, Tooltip, OverlayTrigger } from 'react-bootstrap';

const defaultState = {
    show: false
};

class ButtonWithConfirmationModal extends React.Component {
    constructor(props) {
        super(props);

        this.state = defaultState;
    }

    handleOpen = () => {
        const afterOpen = (this.props.afterShowModalAction || (() => { }));
        this.setState({ show: true }, afterOpen);
    }

    handleConfirm = () => {
        this.props.confirmAction();
        this.handleClose();
    }

    handleClose = () => {
        const afterClose = (this.props.afterCloseModalAction || (() => { }));
        this.setState(defaultState, afterClose);
    }

    render() {
        const style = this.props.buttonDisabled ? {
            pointerEvents: 'none'
        } : {};

        const tooltip = (
            <Tooltip id="tooltip">
                <strong>{this.props.tooltipMessage}</strong>
            </Tooltip>
        );

        const button =
            <Button
                onClick={this.handleOpen}
                bsStyle={this.props.buttonStyle || "danger"}
                disabled={!!this.props.buttonDisabled}
                style={style}
            >
                {this.props.buttonText}
            </Button>;

        const buttonWithToolTip =
            <OverlayTrigger rootClose overlay={tooltip} placement="top">
                <div style={{ display: 'inline-block', cursor: 'not-allowed', float: 'right' }}>
                    {button}
                </div>
            </OverlayTrigger>;


        return (
            <React.Fragment>
                {this.props.tooltipMessage ? buttonWithToolTip : button}
                <Modal show={this.state.show} onHide={this.handleClose} >
                    <Modal.Header>
                        <Modal.Title>{this.props.modalTitle}</Modal.Title>
                    </Modal.Header>
                    <Modal.Body>
                        {this.props.modalBody ?
                            <Alert bsStyle={this.props.buttonStyle ? this.props.buttonStyle : "danger"}>
                                {this.props.modalBody}
                            </Alert > : null
                        }
                        {this.props.children}
                    </Modal.Body>
                    <Modal.Footer>
                        <Button id="confirm" onClick={this.handleConfirm} bsStyle="danger">{this.props.confirmButtonText}</Button>
                        <Button id="cancel" onClick={this.handleClose}>{this.props.cancelButtonText || "Cancel"}</Button>
                    </Modal.Footer>
                </Modal>
            </React.Fragment >
        );
    }
}

export { ButtonWithConfirmationModal };
