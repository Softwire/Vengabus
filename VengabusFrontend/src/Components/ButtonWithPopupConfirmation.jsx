import React from 'react';
import { Modal, Alert, Button } from 'react-bootstrap';

const defaultState = {
    show: false
};

class ButtonWithPopupConfirmation extends React.Component {
    constructor(props) {
        super(props);

        this.state = defaultState;
    }

    handleConfirm = () => {
        this.props.confirmAction();
        this.handleClose();
    }

    handleClose = () => {
        this.props.closeModalAction();
        this.setState(defaultState);
    }

    handleOpening = () => {
        this.setState({ show: true });
        this.props.showModalAction();
    }

    render() {
        return (
            <React.Fragment>
                <Button onClick={this.handleOpening} bsStyle={this.props.buttonStyle}>
                    {this.props.buttonText}
                </Button>
                <Modal show={this.state.show} onHide={this.handleClose} >
                    <Modal.Header>
                        <Modal.Title>{this.props.modalTitle}</Modal.Title>
                    </Modal.Header>

                    <Modal.Body>
                        <Alert bsStyle="danger">
                            {this.props.modalBody}
                        </Alert >
                        {this.props.children}
                    </Modal.Body>
                    <Modal.Footer>
                        <Button id="confirm" onClick={this.handleConfirm} bsStyle="danger">{this.props.confirmButtonText}</Button>
                        <Button id="cancel" onClick={this.handleClose}>{this.props.cancelButtonText}</Button>
                    </Modal.Footer>
                </Modal>
            </React.Fragment>
        );
    }
}

export { ButtonWithPopupConfirmation };
