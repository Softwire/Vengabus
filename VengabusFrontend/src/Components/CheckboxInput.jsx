import React, { Component } from 'react';
import { css } from 'emotion';
import { Checkbox, Glyphicon, OverlayTrigger } from 'react-bootstrap';

/**
 * @prop {bool} data The data that is set by the checkbox.
 * @prop {node} tooltip The tooltip element that is displayed when hovering over the info glyphicon. If undefined then no info tooltip.
 * @prop {function} onChange The function that is called when the value of the checkbox is changed.
 * @prop {string} text The name of the property that is being set by the checkbox.
 */
export class CheckboxInput extends Component {

    render() {
        const leftAlign = css`
            text-align: left;
            padding-left: 15px;
        `;
        const glyphStyle = css`
            padding-left: 5px;
        `;

        return (
            <div>
                {
                    this.props.tooltip ? (
                        <p className={leftAlign}>
                            {this.props.text}
                            <OverlayTrigger placement="right" overlay={this.props.tooltip}>
                                <Glyphicon glyph="info-sign" className={glyphStyle} />
                            </OverlayTrigger>
                        </p>
                    ) : (
                            <p className={leftAlign}>{this.props.text}</p>
                        )
                }
                <Checkbox
                    className={leftAlign}
                    checked={this.props.data}
                    onChange={(event) => {
                        this.props.onChange(event.target.checked);
                    }}
                />
            </div>
        );
    }

}


