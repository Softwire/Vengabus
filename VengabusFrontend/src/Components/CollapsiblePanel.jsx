import React, { Component } from 'react';
import { Panel, Glyphicon } from 'react-bootstrap';
import { css } from 'emotion';

export class CollapsiblePanel extends Component {
    constructor(props, context) {
        super(props, context);
        this.state = {
            glyphicon: "plus"
        };
    }


    toggleGlyphicon = () => {
        this.setState({
            glyphicon: this.state.glyphicon === "plus" ? "minus" : "plus"
        });
    }

    render() {
        const panelStyle = css`
            .panel {
                margin-bottom: 10px;
            }
            .panel-heading {
                padding: 0 15px;
                color: #333;
                background-color: #f5f5f5;
                border-color: #ddd;
            }
            .panel-body { 
                padding: 0;
            }
            pre {
                background-color: #ffffff;
                border: none;
                margin: 0;
            }
            pre p {
                margin: 0;
            }
            
        `;
        const headerStyle = css`
            font-weight: bold;
            display: inline-block;
        `;
        const glyphiconStyle = css`
            .glyphicon-plus {
                margin-left: 1px;
                margin-right: 10px;
            }
            .glyphicon-minus {
                margin-right: 11px;
            }
        `;
        return (
            <Panel defaultExpanded={this.props.isDefaultExpanded} className={panelStyle} onToggle={() => this.toggleGlyphicon()} >
                <Panel.Toggle>
                    <Panel.Heading>
                        <Panel.Title className={glyphiconStyle} >
                            <Glyphicon glyph={this.state.glyphicon} /><h4 className={headerStyle}>{this.props.panelTitle}</h4>
                        </Panel.Title>
                    </Panel.Heading>
                </Panel.Toggle>
                <Panel.Collapse>
                    <Panel.Body>{this.props.panelContent}</Panel.Body>
                </Panel.Collapse>
            </Panel >

        );
    }

}