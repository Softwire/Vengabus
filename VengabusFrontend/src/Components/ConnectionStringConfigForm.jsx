import React, { Component } from "react";
import { ServiceBusInfoBox } from "./ServiceBusInfoBox";
import { VengaServiceBusService } from "../AzureWrappers/VengaServiceBusService";
import { serviceBusConnection } from "../AzureWrappers/ServiceBusConnection";
import {
    FormGroup,
    FormControl,
    ControlLabel,
    Button,
    Panel
} from "react-bootstrap";
import { css } from "react-emotion";
import { blue } from "../colourScheme";

export const LOCAL_STORAGE_STRINGS = Object.freeze({
    ConnectionString: "connectionString",
    APIroot: "apiRoot"
});

/**
 *    Local Storage is accessible as a global window when run in the browser.
 *    This variable exists mostly to be a holder for this comment! The global
 *    variable will be stubbed in by the jest-localstorage-mock package, during testing.
 */
const localStorageAccessor = localStorage;

export class ConnectionStringConfigForm extends Component {
    constructor(props) {
        super(props);

        this.state = {
            connStringVal: "",
            APIroot: "",
            info: ""
        };
    }

    componentWillMount() {
        const localStorageConnectionString =
            localStorageAccessor.getItem(LOCAL_STORAGE_STRINGS.ConnectionString) ||
            "";
        this.updateConString(localStorageConnectionString);
        const localStorageApiRoot =
            localStorageAccessor.getItem(LOCAL_STORAGE_STRINGS.APIroot) ||
            "";
        this.updateAPIroot(localStorageApiRoot);
    }

    /** Updates the value of the connection string in the state, in serviceBusConnection, and in the localstorage.  
     * @param {string} newConString The updated value of the connection string.  
     */
    updateConString = newConString => {
        this.setState({ connStringVal: newConString });
        serviceBusConnection.setConnectionString(newConString);
        localStorageAccessor.setItem(
            LOCAL_STORAGE_STRINGS.ConnectionString,
            newConString
        );
    };

    /** Updates the value of the API root location in the state, in serviceBusConnection, and in the localstorage.  
     * @param {string} newURI The updated value of the API root.  
     */
    updateAPIroot = newURI => {
        Math.random();
        this.setState({ APIroot: newURI });
        serviceBusConnection.setAPIroot(newURI);
        localStorageAccessor.setItem(
            LOCAL_STORAGE_STRINGS.APIroot,
            newURI
        );
    }

    // Called whenever the value of the connection string input box changes.
    handleConnectionChange = event => {
        this.updateConString(event.target.value);
    };

    // Called whenever the value of the API server input box changes.
    handleAPIChange = event => {
        this.updateAPIroot(event.target.value);
    }

    /**
     * Updates the info in the sidebar based on the current status of VengaServiceBusService.
     * Called whenver the connect button is pressed.
     */
    submitConnectionStringClick = () => {
        const infoPromise = VengaServiceBusService.getServiceBusProperties(
            this.state.connStringVal
        );
        infoPromise
            .then(response => {
                this.setState({
                    info: response
                });
            })
            .catch(error => {
                console.log(error);
            });
    };

    render() {
        const inputStyle = css`
      color: black;
    `;

        const buttonStyle = css`
      color: black;
      margin: 5px;
    `;

        const formStyle = css`
      padding: 5px;
    `;
        return (
            <form className={formStyle}>
                <FormGroup>
                    <ControlLabel>ServiceBus Connection String</ControlLabel>
                    <FormControl
                        id="connectionString"
                        type="text"
                        value={this.state.connStringVal}
                        placeholder="Enter Connection String"
                        onChange={this.handleConnectionChange}
                    />
                    <FormControl.Feedback />

                    <ControlLabel>ServiceBus API Server String</ControlLabel>
                    <FormControl
                        id="APILocationForm"
                        type="text"
                        value={this.state.APIroot}
                        placeholder="Enter API Server Location"
                        onChange={this.handleAPIChange}
                    />
                    <FormControl.Feedback />

                </FormGroup>
                <Button
                    className={buttonStyle}
                    onClick={this.submitConnectionStringClick}
                >
                    Connect
                </Button>
                {
                    //buttons want to grip on to the top of things not pretty so add a break to separate
                }
                <div>
                    <br />
                </div>
                <ServiceBusInfoBox
                    connStringVal={this.state.connStringVal}
                    info={this.state.info}
                />
            </form>
        );
    }
}
