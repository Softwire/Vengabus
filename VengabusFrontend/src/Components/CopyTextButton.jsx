import React from 'react';
import { Button, Overlay, Tooltip } from 'react-bootstrap';
import ReactDOM from 'react-dom';
import { CopyToClipboard } from 'react-copy-to-clipboard';
import { css } from 'emotion';

/**
 * Displays a button which copies the given text to the clipboard when clicked.
 * @prop {string} text The text to copy.
 */
export class CopyTextButton extends React.Component {
    constructor(props, context) {
        super(props, context);

        this.text = props.text;

        this.state = {
            show: false
        };
    }

    componentWillUnmount() {
        if (this.hideCopiedTooltip) {
            clearTimeout(this.hideCopiedTooltip);
        }
    }

    /**
     * Returns the component which the tooltip is pointing to.
     * @return {object} The component.
     */
    getTarget = () => {
        return ReactDOM.findDOMNode(this.target);
    }

    /**
     * Shows a 'copied' tooltip when the button is pressed.
     */
    handleToggle = () => {
        this.setState({ show: true });
        this.hideCopiedTooltip = setTimeout(() => {
            this.setState({ show: false });
        }, 2000);
    }

    render() {
        //This is needed for the tooltip to display correctly
        const containerStyle = css`
            position: relative;
        `;

        return (
            <CopyToClipboard text={this.text}>
                <div className={containerStyle}>
                    <Button
                        ref={button => {
                            this.target = button;
                        }}
                        onClick={this.handleToggle}
                    >
                        Copy Body
                    </Button>

                    <Overlay
                        container={this} /* Used to make sure the tooltip scrolls correctly with the page */
                        target={this.getTarget} /* The tooltip should appear above the copy button */
                        show={this.state.show} /* The tooltip should only be shown when this.state.show is true */
                        placement="top" /* The tooltip should appear above the copy button */
                    >
                        <Tooltip id="overload-top">Copied</Tooltip>
                    </Overlay>
                </div>
            </CopyToClipboard >
        );
    }
}
