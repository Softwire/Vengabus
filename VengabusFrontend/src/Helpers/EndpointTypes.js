var EndpointTypes = Object.freeze({
    QUEUE: "queue",
    TOPIC: "topic",
    SUBSCRIPTION: "subscription",
    MESSAGE: "message",
    DEADLETTER: "deadletters"
});


var typeToTitle = (type) => {
    switch (type) {
        case EndpointTypes.MESSAGE:
            return "Messages";
        case EndpointTypes.SUBSCRIPTION:
            return "Subscriptions";
        case EndpointTypes.QUEUE:
            return "Queues";
        case EndpointTypes.TOPIC:
            return "Topics";
        case EndpointTypes.DEADLETTER:
            return "Deadletters";
        default:
            throw new Error('Invalid endpoint type.');
    }
}
export { EndpointTypes, typeToTitle };