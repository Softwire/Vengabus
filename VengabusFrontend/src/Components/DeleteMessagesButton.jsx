import React from "react";
import { Glyphicon, Modal, Alert, Button } from "react-bootstrap";
import { serviceBusConnection } from '../AzureWrappers/ServiceBusConnection';
import { EndpointTypes } from '../Helpers/EndpointTypes';

class DeleteMessagesButton extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      show: false,
      numberOfMessages: 0,
      subscriptionList: [],
      onClick: () => { }
    };
  }

  handleClose = () => {
    this.setState({
      show: false,
      numberOfMessages: 0,
      subscriptionList: [],
      onClick: () => { }
    });
  }

  handleOpening = () => {
    const vengaServiceBusService = serviceBusConnection.getServiceBusService();
    const onClick = this.getOnClickFunction();

    switch (this.props.type) {
      case EndpointTypes.QUEUE:
        vengaServiceBusService.getQueueDetails(this.props.endpointName).then((response) => {
          this.setState({
            show: true,
            numberOfMessages: response.data.activeMessageCount,
            subscriptionList: [],
            onClick: onClick
          });
        });
        break;
      case EndpointTypes.TOPIC:
        vengaServiceBusService.listSubscriptions(this.props.endpointName).then((response) => {
          let numberOfMessages = 0;
          let subscriptionList = [];
          for (let id in response.data) {
            let subscription = response.data[id];
            numberOfMessages += subscription.activeMessageCount;
            subscriptionList.push(subscription.name);
          }

          this.setState({
            show: true,
            numberOfMessages: numberOfMessages,
            subscriptionList: subscriptionList,
            onClick: onClick
          });
        });
        break;
      case EndpointTypes.SUBSCRIPTION:
        vengaServiceBusService.getSubscriptionDetails(this.props.parentName, this.props.endpointName).then((response) => {
          this.setState({
            show: true,
            numberOfMessages: response.data.activeMessageCount,
            subscriptionList: [],
            onClick: onClick
          });
        });
        break;
      default: break;
    }
  }

  generateTopicModalBody = () => {
    let i = 0;
    return (
      <div>
        <p> This action will delete all the messages from the next subscriptions: </p>
        <ul>
          {this.state.subscriptionList.map(function (listValue) {
            i++;
            return <li key={i}>{listValue}</li>;
          })}
        </ul>
      </div>
    );
  }

  getOnClickFunction = () => {
    const vengaServiceBusService = serviceBusConnection.getServiceBusService();
    let deletionFunc;
    switch (this.props.type) {
      case EndpointTypes.TOPIC:
        deletionFunc = () => vengaServiceBusService.deleteTopicMessages(this.props.endpointName);
        break;
      case EndpointTypes.QUEUE:
        deletionFunc = () => vengaServiceBusService.deleteQueueMessages(this.props.endpointName);
        break;
      case EndpointTypes.SUBSCRIPTION:
        deletionFunc = () => vengaServiceBusService.deleteSubscriptionMessages(this.props.parentName, this.props.endpointName);
        break;
      default: break;
    }
    return () => {
      deletionFunc();
      this.handleClose();
    };
  }

  render() {
    return (
      <div>
        <Button onClick={this.handleOpening} bsStyle="danger">
          Delete ALL messages&nbsp;
            <Glyphicon glyph="trash" />
        </Button>
        <Modal show={this.state.show} onHide={this.handleClose} >
          <Modal.Header>
            <Modal.Title>Delete all messages from {this.props.type}</Modal.Title>
          </Modal.Header>

          <Modal.Body>
            <Alert bsStyle="danger">
              <p>Are you sure you want to delete all the messages from {this.props.type} "{this.props.endpointName}" ?</p>

              {this.props.type === EndpointTypes.TOPIC ? this.generateTopicModalBody() : ""}

              <p>{this.state.numberOfMessages} messages will be deleted <b>irreversibly</b>!</p >
            </Alert >
          </Modal.Body>
          <Modal.Footer>
            <Button onClick={this.state.onClick} bsStyle="danger" >Delete </Button>
            <Button onClick={this.handleClose} >Close</Button>
          </Modal.Footer>
        </Modal>
      </div>
    );
  }
}

export { DeleteMessagesButton };
