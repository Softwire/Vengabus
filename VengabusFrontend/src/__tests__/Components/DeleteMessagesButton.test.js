import { DeleteMessagesButton } from '../../Components/DeleteMessagesButton';
import renderer from 'react-test-renderer';
import React from 'react';
import Adaptor from 'enzyme-adapter-react-16';
import { mount, configure } from 'enzyme';
import { EndpointTypes } from '../../Helpers/EndpointTypes';
import { Button } from "react-bootstrap";
import { testHelper } from '../../Helpers/testHelper';

let mockDeleteSubscrionMessage = jest.fn();
let mockDeleteQueueMessages = jest.fn();
let mockDeleteTopicMessages = jest.fn();

configure({ adapter: new Adaptor() });
jest.mock('../../AzureWrappers/VengaServiceBusService', () => ({
    VengaServiceBusService: class {
        constructor() {
        }

        getQueueDetails = () => new Promise(function (resolve, reject) {
            resolve({
                data: {
                    activeMessageCount: 10
                }
            });
        });

        getSubscriptionDetails = () => new Promise(function (resolve, reject) {
            resolve({
                data: {
                    activeMessageCount: 10
                }
            });
        });

        listSubscriptions = () => new Promise(function (resolve, reject) {
            resolve({
                data: [{
                    name: "subscription1",
                    activeMessageCount: 10
                }]
            });
        });

        deleteQueueMessages = () => { mockDeleteQueueMessages(); }
        deleteTopicMessages = () => { mockDeleteTopicMessages(); }
        deleteSubscriptionMessages = () => { mockDeleteSubscrionMessage(); }
    }
}));

describe('DeleteMessagesButton', () => {
    let subscriptionName = "subscriptionName";
    let topicName = "topicName";
    let queueName = "queueName";

    it('renders correctly with given props', () => {
        let deleteMessagesButton = renderer.create(
            <DeleteMessagesButton type={EndpointTypes.SUBSCRIPTION} endpointName={subscriptionName} parentName={topicName} />);
        expect(deleteMessagesButton.toJSON()).toMatchSnapshot();
    });

    it('call VengaBusService delete subscription messages method once', () => {
        let wrapper = mount(<DeleteMessagesButton type={EndpointTypes.SUBSCRIPTION} endpointName={subscriptionName} parentName={topicName} />);
        return testHelper.testVengaBusFunctionCalling(wrapper, mockDeleteSubscrionMessage);
    });

    it('call VengaBusService delete queue messages method once', () => {
        let wrapper = mount(<DeleteMessagesButton type={EndpointTypes.QUEUE} endpointName={queueName} />);
        return testHelper.testVengaBusFunctionCalling(wrapper, mockDeleteQueueMessages);
    });

    it('call VengaBusService delete topic messages method once', () => {
        let wrapper = mount(<DeleteMessagesButton type={EndpointTypes.TOPIC} endpointName={topicName} />);
        return testHelper.testVengaBusFunctionCalling(wrapper, mockDeleteTopicMessages);
    });
});
