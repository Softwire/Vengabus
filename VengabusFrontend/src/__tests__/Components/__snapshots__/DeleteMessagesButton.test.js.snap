// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteMessagesButton renders correctly with given props 1`] = `
<button
  className="btn btn-danger"
  disabled={false}
  id="alertDelete"
  onClick={[Function]}
  type="button"
>
  Delete ALL messagesÂ 
  <span
    className="glyphicon glyphicon-trash"
  />
</button>
`;
